src/TesApi.Web/Startup.cs

```diff
@@ -107,6 +108,9 @@ public void ConfigureServices(IServiceCollection services)
                     .AddSingleton<ConfigurationUtils>()
                     .AddSingleton<IAllowedVmSizesService, AllowedVmSizesService>()
                     .AddSingleton<TokenCredential>(s => new DefaultAzureCredential())
+                    .AddSingleton<TaskToNodeTaskConverter>()
+                    .AddSingleton<TaskExecutionScriptingManager>()
+                    .AddTransient<BatchNodeScriptBuilder>()
```

Comment:
Isn't this a utility class?  Is it really appropriate for dependency injection?

Reply:
hmm, not sure I agree. These are dependencies that are required to be injected and their creation requires options and other dependencies from the app context.  I think DI is doing what is supposed to be do.